esti/gc_test.go

```diff
@@ -0,0 +1,255 @@
+	if err != nil {
+		t.Fatal("Set GC rules", err)
+	}
+
+	for _, branch := range testCase.branches {
+		if branch.deleteCommitDaysAgo > -1 {
+			_, err = client.DeleteObjectWithResponse(ctx, repo, branch.name, &api.DeleteObjectParams{Path: "file" + testCase.id})
+			epochCommitDateInSeconds := currentEpochInSeconds - (dayInSeconds * branch.deleteCommitDaysAgo)
+			_, err = client.CommitWithResponse(ctx, repo, branch.name, &api.CommitParams{}, api.CommitJSONRequestBody{Message: "Deleted file" + testCase.id, Date: &epochCommitDateInSeconds})
+			if err != nil {
+				t.Fatal("Commit some data", err)
+			}
+			_, _ = uploadFileRandomData(ctx, t, repo, branch.name, "file"+testCase.id+"not_deleted", false)
+			// This is for the previous commit to be the HEAD of the branch outside the retention time (according to GC https://github.com/treeverse/lakeFS/issues/1932)
+			_, err = client.CommitWithResponse(ctx, repo, branch.name, &api.CommitParams{}, api.CommitJSONRequestBody{Message: "not deleted file commit: " + testCase.id, Date: &epochCommitDateInSeconds})
+			if err != nil {
+				t.Fatal("Commit some data", err)
+			}
+		} else {
+			_, err = client.DeleteBranchWithResponse(ctx, repo, branch.name)
+			if err != nil {
+				t.Fatal("Delete brach", err)
+			}
+		}
+	}
+	return commitId
+}
+
+func getGCMode(testCase testCase) int {
+	if testCase.testMode == "mark" {
+		return markOnlyMode
+	} else if testCase.testMode == "sweep" {
+		return sweepOnlyMode
+	} else {
+		return fullGCMode
+	}
+}
+
+func runGC(repo string, gcMode int, markId string, awsAccessKey string, awsSecretKey string) {
+	sweepConfig := ""
+	markIDConfig := ""
+	if gcMode == markOnlyMode || gcMode == sweepOnlyMode {
+		// If we are in sweep-only mode, we'll first mark and afterwards we'll sweep
+		sweepConfig = "-c spark.hadoop.lakefs.gc.do_sweep=false "
+	}
+	if markId != "" {
+		markIDConfig = "-c spark.hadoop.lakefs.gc.mark_id=" + markId + " "
+	}
+
+	cmd := exec.Command("/bin/sh", "-c",
+		"cd target; CLIENT_JAR=$(ls core-301/scala-2.12/lakefs-spark-client-301-assembly-*.jar); cd ..; echo $CLIENT_JAR;\n"+
+			"docker run --network host --add-host lakefs:127.0.0.1 -v $(pwd)/target:/local -v $(pwd)/ivy:/opt/bitnami/spark/.ivy2 --rm docker.io/bitnami/spark:3.2 spark-submit\n"+
+			"--master spark://localhost:7077\n"+
+			"--conf spark.hadoop.lakefs.api.url=http://lakefs:8000/api/v1\n"+
+			sweepConfig+markIDConfig+
+			"--conf spark.hadoop.fs.s3.impl=org.apache.hadoop.fs.s3a.S3AFileSystem\n"+
+			"--conf spark.hadoop.fs.s3a.access.key="+awsAccessKey+"\n"+
+			"--conf spark.hadoop.fs.s3a.secret.key="+awsSecretKey+"\n"+
+			"--conf spark.hadoop.lakefs.api.access_key=AKIAIOSFDNN7EXAMPLEQ\n"+
+			"--conf spark.hadoop.lakefs.api.secret_key=wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY\n"+
+			"--class io.treeverse.clients.GarbageCollector\n"+
+			"--packages org.apache.hadoop:hadoop-aws:2.7.7\n"+
+			"/local/${CLIENT_JAR}\n"+
+			repo+" us-east-1")
```

Comment:
nit: I would extract this and the sweep-only-mode below into two functions that format the commands.